R/system.R

```diff
@@ -2824,7 +2824,13 @@ read_excel_file <- function(path, sheet = 1, col_names = TRUE, col_types = NULL,
     }
     if(!is.null(tzone)) { # if timezone is specified, apply the timezeon to POSIXct columns
       df <- df %>% dplyr::mutate_if(lubridate::is.POSIXct, funs(lubridate::force_tz(., tzone=tzone)))
+      # make sure to convert POSIXct to Date if the col_types are set as Date specifically
+      df <- df %>% dplyr::mutate_at(vars(colnames(df)[col_types == "date"]), funs(as.Date(.,tz=tzone)))
```

Comment:
Look into the behavior of readxl, and make sure setting timezone would not make the result different from the original date.